isthmus/src/test/java/io/substrait/isthmus/RelCopyOnWriteVisitorTest.java

```diff
@@ -105,6 +103,29 @@ public void replaceCountDistincts() throws IOException, SqlParseException {
     assertPlanRoundrip(newPlan);
   }
 
+  @Test
+  public void approximateCountDistinct() throws IOException, SqlParseException {
+    Plan oldPlan =
+        buildPlanFromQuery(
+            "select count(distinct l_discount), count(distinct l_tax) from lineitem");
+    assertEquals(2, new CountCountDistinct().getCountDistincts(oldPlan));
+    assertEquals(0, new CountApproxCountDistinct().getApproxCountDistincts(oldPlan));
+    ReplaceCountDistinctWithApprox action = new ReplaceCountDistinctWithApprox();
+    Plan newPlan = action.modify(oldPlan).orElse(oldPlan);
+    assertEquals(2, new CountApproxCountDistinct().getApproxCountDistincts(newPlan));
+    assertEquals(0, new CountCountDistinct().getCountDistincts(newPlan));
+    assertPlanRoundrip(newPlan);
+
+    // convert newPlan back to sql
+    var pojoRel = newPlan.getRoots().get(0).getInput();
+    String[] values = asString("tpch/schema.sql").split(";");
+    var creates = Arrays.stream(values).filter(t -> !t.trim().isBlank()).toList();
+    RelNode relnodeRoot = new SubstraitToSql().substraitRelToCalciteRel(pojoRel, creates);
+    String newSql = SubstraitToSql.toSql(relnodeRoot);
+    assertTrue(newSql.toUpperCase().contains("APPROX_COUNT_DISTINCT"));
```

Comment:
Stronger assert would be good her. Couple of things here would be nice:
- Compare equality of original SQL where you replace `count(distinct <col>)` with `approx_count_distinct(<col>)`
- Convert newSql back to Plan using buildPlanFromQuery and then to proto, and compare to newPlan converted to proto.

Reply:
For the first equality checking of SQL:
   - I feel comparing text is not a very stable way, since any format changing in Calcite's  RelToSql class could lead to testcase failure. 
   - Also, the converted `newSQL` contains a different field name `$f0` for the expression. That issue will be addressed later on.  Before that is fixed, could not compare SQL directly. 
 
For the second checking of substrait pojo/proto plan : 
- This pr adds support of `approx_count_distinct()` in substrait rel -> sql.  But the other direction (`sql` -> `substrait rel`) does not have support of `approx_count_distinct()`.  That probably need a separate PR, as I feel it's not straightforward to fix it (requires changing the function resolution logic). 
- So, for now, I could not add the checking of pojo/proto plan.  I put TODO to add new checking once 'sql' -> substrait work properly. 
-